lib/scarpe/wv/shape_helper.rb

```diff
@@ -18,8 +18,23 @@ def line_to(x, y)
   def shape_path
     path_commands_str = path_commands.join(" ")
     path_commands_str
+ 
   end
 
+  def colors
+    $colors ||= []
+  end
+
+  def fill(color)
+     colors << [color]
+  end
+
+
+  def color_for_fill
```

Comment:
Huh. So this is popping an entry off the colors stack every time you call it? That seems odd, though I could be misunderstanding.

Reply:
hm yea it is..

Reply:
idk any othr way i can use for code like

```
Shoes.app do
fill "red"
star //red
fill "green"
oval//green
end

```

Reply:
if there is better way i am open to it!  idk any other way so i am just trying things arnd

Reply:
So, there's a standard-ish way to handle this kind of thing, sometimes called a "graphics context". Basically you have a structure with all the changeable style settings like stroke, fill, font size, etc. And so you change that graphics context as you go along, and you read the graphics context when you need a current fill, etc. Which is fine.

The problem with pop-on-read isn't that you have a context, mutable state, etc. Having a context is just fine. The problem with "fill does a push and reading does a pop" is stuff like this:

```
  shape do
    fill "green"
    fill "red"
    star; star; star
    fill "yellow"
    star; star; star
  end
```

You want to write the most recent fill into a context and then just read, don't pop. That way if you set a fill like green it'll stay green until you change it (which is good), not go back to some previous fill, which is unexpected.

Reply:
btw this has been updated as per my understanding of what you said , can u do a quick check?